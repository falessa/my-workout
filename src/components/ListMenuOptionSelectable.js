import React from 'react';
import { StyleSheet, View, Text, FlatList } from 'react-native';
import { useTranslation } from 'react-i18next';
import { TouchableOpacity } from 'react-native-gesture-handler';
import Languages from '../utils/LanguagesEnum'
import i18n from 'i18next'

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        alignItems: 'stretch',
        justifyContent: 'center',
        width: '95%',
        borderBottomWidth: 1,
        borderBottomColor: '#eee',
        padding: 10
    },
    text: {
        fontSize: 18,
        // paddingBottom: 10,
    }
})

const ListMenuOptionSelectable = ({ options }) => {
    const { t } = useTranslation()
    
    const changeLanguage = (language) => {
        i18n.changeLanguage(language)
    }

    const selectMenuOption = item => {
        switch(item) {
            case Languages.ENGLISH: {
                changeLanguage('en')
                break
            }
            case Languages.SPANISH: {
                changeLanguage('es')
                break
            }
            default: {
                throw new Error(`Language ${item} does not exist`)
            }
        }
    }

    return (
        <View style={styles.container}>
            <FlatList
                data={options}
                keyExtractor={x => String(x)}
                renderItem={({ item }) => 
                    <TouchableOpacity style={styles.container} onPress={() => selectMenuOption(item)}>
                        <Text style={styles.text}>{t(item)}</Text>
                    </TouchableOpacity>
                }
            />
        </View>
    )
}

export default ListMenuOptionSelectable;