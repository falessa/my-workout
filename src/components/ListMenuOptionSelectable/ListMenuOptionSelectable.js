import React from 'react';
import { useSelector, useDispatch } from 'react-redux'
import { setLanguage } from '../../redux/settings'
import { StyleSheet, View, Text, FlatList } from 'react-native';
import { useTranslation } from 'react-i18next';
import { TouchableOpacity } from 'react-native-gesture-handler';
import { Ionicons } from '@expo/vector-icons';
import Languages from '../../utils/LanguagesEnum'
import i18n from 'i18next'

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        alignItems: 'stretch',
        justifyContent: 'center',
        width: '95%',
        borderBottomWidth: 1,
        borderBottomColor: '#eee',
        padding: 10
    },
    text: {
        fontSize: 18,
        // padding: 10
    },
    optionSelected: {
        fontSize: 18,
        fontWeight: 'bold',
        paddingRight: '75%',
        // color: '#386fa4' // when applying the color, the colored text is not displayed on android devices
    }
})

const ListMenuOptionSelectable = ({ options }) => {
    const dispatch = useDispatch();
    const settings = useSelector(store => store.settings)
    const languageStored = settings.language

    const { t } = useTranslation()

    const changeLanguage = (language) => {
        i18n.changeLanguage(language)
    }

    const selectMenuOption = item => {
        switch (item) {
            case Languages.ENGLISH: {
                changeLanguage('en')
                dispatch(setLanguage(Languages.ENGLISH))
                break
            }
            case Languages.SPANISH: {
                changeLanguage('es')
                dispatch(setLanguage(Languages.SPANISH))
                break
            }
            default: {
                throw new Error(`Language ${item} does not exist`)
            }
        }
    }

    return (
        <View style={styles.container}>
            <FlatList
                data={options}
                keyExtractor={x => String(x)}
                renderItem={({ item }) =>
                    <TouchableOpacity style={styles.container} onPress={() => selectMenuOption(item)}>
                        <View style={{ flexDirection: 'row' }}>
                            <Text style={languageStored === item ? styles.optionSelected : styles.text}>{t(item)}</Text>
                            {languageStored === item
                                ? <Ionicons name='checkmark' size={22} color={'#386fa4'} />
                                : <></>
                            }
                        </View>
                    </TouchableOpacity>
                }
            />
        </View>
    )
}

export default ListMenuOptionSelectable;