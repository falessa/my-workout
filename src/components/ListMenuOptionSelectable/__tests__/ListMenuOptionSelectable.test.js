import "react-native";
import React from "react";
import { create } from "react-test-renderer";
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import ListMenuOptionSelectable from '../ListMenuOptionSelectable';

describe('ListMenuOptionSelectable tests', () => {
    const props = {
        options: [
            "Selectable Option 1",
            "Selectable Option 2",
            "Selectable Option 3"
        ]
    };

    const mockStore = configureStore([]);
    const store = mockStore({
        settings: {
            language: 'en'
        }
    });

    const wrapper = create(
        <Provider store={store}>
            <ListMenuOptionSelectable {...props} />
        </Provider>
    );

    test('Render ListMenuOptionSelectable', () => {
        expect(wrapper).toMatchSnapshot()
    });
});
