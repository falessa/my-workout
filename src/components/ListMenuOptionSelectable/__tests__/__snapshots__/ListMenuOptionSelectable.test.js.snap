// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListMenuOptionSelectable tests Render ListMenuOptionSelectable 1`] = `
<View
  style={
    {
      "alignItems": "stretch",
      "backgroundColor": "#fff",
      "borderBottomColor": "#eee",
      "borderBottomWidth": 1,
      "flex": 1,
      "justifyContent": "center",
      "padding": 10,
      "width": "95%",
    }
  }
>
  <RCTScrollView
    data={
      [
        "Selectable Option 1",
        "Selectable Option 2",
        "Selectable Option 3",
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <RNGestureHandlerButton
          collapsable={false}
          delayLongPress={600}
          enabled={true}
          exclusive={true}
          handlerTag={1}
          handlerType="NativeViewGestureHandler"
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          rippleColor={0}
          touchSoundDisabled={false}
        >
          <View
            accessible={true}
            collapsable={false}
            style={
              {
                "alignItems": "stretch",
                "backgroundColor": "#fff",
                "borderBottomColor": "#eee",
                "borderBottomWidth": 1,
                "flex": 1,
                "justifyContent": "center",
                "opacity": 1,
                "padding": 10,
                "width": "95%",
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  {
                    "fontSize": 18,
                  }
                }
              >
                Selectable Option 1
              </Text>
            </View>
          </View>
        </RNGestureHandlerButton>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <RNGestureHandlerButton
          collapsable={false}
          delayLongPress={600}
          enabled={true}
          exclusive={true}
          handlerTag={2}
          handlerType="NativeViewGestureHandler"
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          rippleColor={0}
          touchSoundDisabled={false}
        >
          <View
            accessible={true}
            collapsable={false}
            style={
              {
                "alignItems": "stretch",
                "backgroundColor": "#fff",
                "borderBottomColor": "#eee",
                "borderBottomWidth": 1,
                "flex": 1,
                "justifyContent": "center",
                "opacity": 1,
                "padding": 10,
                "width": "95%",
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  {
                    "fontSize": 18,
                  }
                }
              >
                Selectable Option 2
              </Text>
            </View>
          </View>
        </RNGestureHandlerButton>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <RNGestureHandlerButton
          collapsable={false}
          delayLongPress={600}
          enabled={true}
          exclusive={true}
          handlerTag={3}
          handlerType="NativeViewGestureHandler"
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          rippleColor={0}
          touchSoundDisabled={false}
        >
          <View
            accessible={true}
            collapsable={false}
            style={
              {
                "alignItems": "stretch",
                "backgroundColor": "#fff",
                "borderBottomColor": "#eee",
                "borderBottomWidth": 1,
                "flex": 1,
                "justifyContent": "center",
                "opacity": 1,
                "padding": 10,
                "width": "95%",
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  {
                    "fontSize": 18,
                  }
                }
              >
                Selectable Option 3
              </Text>
            </View>
          </View>
        </RNGestureHandlerButton>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
